### PR TITLE
[READY] Master fix em num

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #[14337](https://bugs.limesurvey.org/view.php?id=14337): Comparaison String and Numeric is different in same page and other page
Fixed issue #[14198](https://bugs.limesurvey.org/view.php?id=14198): max function broken : diff in JS and PHP, diff with previous behaviour

Still issue with PHP7.2 and number + string, but 
>  Be careful when passing arguments with mixed types values because max() can produce unpredictable results. 
https://php.net/manual/en/function.max.php

max and min from http://locutus.io/php/math/

### CODE REVIEW

**Comment by Shnoulle** [comment]
Move to WIP : issue with -1,-2 etc …

File: assets/scripts/expressions/em_javascript.js [hunk 1]
```diff
@@ -322,89 +322,56 @@ function LEMif(testDone,ifTrue,ifFalse)
...
- * max like php : https://github.com/kvz/locutus/blob/master/src/php/math/max.js
+ * max like php in LimeSurvey, start by https://github.com/kvz/locutus/blob/master/src/php/math/max.js
+ * Review for ExpressionManager
  **/
 function LEMmax () {
   //  discuss at: http://locutus.io/php/max/
   // original by: Onno Marsman (https://twitter.com/onnomarsman)
-  //  revised by: Onno Marsman (https://twitter.com/onnomarsman)
-  // improved by: Jack
+  //  revised by: Denis Chenu for LimeSurey specific
```
Comment (tammoterhark) [inline assets/scripts/expressions/em_javascript.js]: LimeSurvey

File: assets/scripts/expressions/em_javascript.js [hunk 2]
```diff
@@ -439,26 +399,27 @@ function LEMmax () {
...
+ * Review for ExpressionManager
+ * @todo : remove object, fix with empty string
  **/
 function LEMmin () {
-  //  discuss at: http://locutus.io/php/min/
+  //  discuss at: http://locutus.io/php/max/
   // original by: Onno Marsman (https://twitter.com/onnomarsman)
-  //  revised by: Onno Marsman (https://twitter.com/onnomarsman)
-  // improved by: Jack
+  //  revised by: Denis Chenu for LimeSurey specific
```
Comment (tammoterhark) [inline assets/scripts/expressions/em_javascript.js]: LimeSurvey

**Comment by Shnoulle** [comment]
Still have issue with min and isNaN(current) && !isNaN(next)

**Comment by Shnoulle** [comment]
OK,

Fixed but : PHP version can throw different value in PHP

Currently : with survey in https://bugs.limesurvey.org/view.php?id=14198#c52303 show the same things in JS and PHP. Can bnot be sure since «e careful when passing arguments with mixed types values because min() can produce unpredictable results. »

**Comment by Shnoulle** [comment]
Travis test survey show error …

**Comment by Shnoulle** [comment]
@tammoterhark : if you have time to make more test , i check on PHP5.6 in some minutes.

But if there are diff for PHP version : we can't fix it (only can put it at information …)

**Comment by tammoterhark** [comment]
No more remarks!

**Comment by Shnoulle** [comment]
No other test todo :'(

**Comment by Shnoulle** [comment]
I close this one to separate the 2 fixes …

1. comparaison
2. min/max

Maybe i need comparaison for min/max :).

**Pull request closed without merging**